CMakeLists.txt

```diff
@@ -45,39 +45,50 @@ set_target_properties(${PROJECT_NAME} PROPERTIES DEBUG_POSTFIX "_debug")
 set(EZC3D_BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
 
 # Create the configuration header file associated with the asked options
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/include/ezc3dConfig.h.in ${EZC3D_BINARY_DIR}/include/ezc3dConfig.h)
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/include/ezc3dConfig.h.in
+        ${EZC3D_BINARY_DIR}/include/ezc3dConfig.h)
 
 # Add headers
-target_include_directories(${PROJECT_NAME} PUBLIC
-    ${CMAKE_CURRENT_SOURCE_DIR}/include
-    ${EZC3D_BINARY_DIR}/include
-)
+# Include directories when building the library:
+target_include_directories(${PROJECT_NAME} PRIVATE
+        "${CMAKE_CURRENT_SOURCE_DIR}/include"
+        "${EZC3D_BINARY_DIR}/include"
+        )
 
 # Install target
 if(WIN32)
     set(${PROJECT_NAME}_LIB_FOLDER Lib)
     set(${PROJECT_NAME}_BIN_FOLDER bin)
     set(${PROJECT_NAME}_INCLUDE_FOLDER include/${PROJECT_NAME})
 else()
-    set(${PROJECT_NAME}_LIB_FOLDER ${CMAKE_INSTALL_PREFIX}/lib/${PROJECT_NAME})
-    set(${PROJECT_NAME}_BIN_FOLDER ${CMAKE_INSTALL_PREFIX}/lib/${PROJECT_NAME})
-    set(${PROJECT_NAME}_INCLUDE_FOLDER ${CMAKE_INSTALL_PREFIX}/include/${PROJECT_NAME})
+    set(${PROJECT_NAME}_LIB_FOLDER lib/${PROJECT_NAME})
+    set(${PROJECT_NAME}_BIN_FOLDER lib/${PROJECT_NAME})
+    set(${PROJECT_NAME}_INCLUDE_FOLDER include/${PROJECT_NAME})
```

Comment:
RPath is no longer set properly for the Python interface if ${CMAKE_INSTALL_PREFIX} is not present. On my computer I get
`Set runtime path of "/home/pariterre/Programmation/miniconda3/envs/ezc3d-dev/lib/python3.8/site-packages/ezc3d/_ezc3d.so" to "lib/ezc3d"`
which is not the right path (unless you change the LD_LIBRARY_PATH which I don't want to do).

Any suggestions?

Reply:
There are two possibilities for the RPATH for python bindings:

A. If the python bindings are expected to be used from their installed location, then you can use a relative RPATH, because you know the relative location of the ezc3d library.

B. If the python bindings are expected to be installed into your python distribution, then you need to use an absolute RPATH to the ezc3d library.

Which do you prefer? I think OpenSim has a CMake flag to toggle between A and B. We have spent a lot of time getting RPATH to work well for OpenSim, so we can refer to what OpenSim does.

Reply:
I think I prefer the case B since I install the python binding in the Python site-package folder.
Unless it makes it incompatible with OpenSim, lets keep it as B :)

Reply:
Okay I pushed a change that fixed the RPATH to be absolute again.